code/game/objects/items/defibrillator.dm

```diff
@@ -128,215 +136,193 @@
 	stack_trace("Powercell deleted while powering the defib, this isn't supposed to happen normally.")
 	set_dcell(null)
 
-/mob/living/proc/get_ghost(bypass_client_check = FALSE)
-	if(client) //Let's call up the correct ghost!
-		return null
-	for(var/mob/dead/observer/ghost AS in GLOB.observer_list)
-		if(!ghost) //Observers hard del often so lets just be safe
-			continue
-		if(isnull(ghost.can_reenter_corpse))
-			continue
-		if(ghost.can_reenter_corpse.resolve() != src)
-			continue
-		if(ghost.client || bypass_client_check)
-			return ghost
-	return null
-
-/mob/living/carbon/human/proc/has_working_organs()
-	var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
-
-	if(!heart || heart.organ_status == ORGAN_BROKEN || !has_brain())
-		return FALSE
+/obj/item/defibrillator/attack(mob/living/carbon/human/patient, mob/living/carbon/human/user)
+	defibrillate(patient,user)
 
+///Proc for checking that the defib is ready to operate
+/obj/item/defibrillator/proc/defib_ready(mob/living/carbon/human/patient, mob/living/carbon/human/user)
+	if(!ready)
+		to_chat(user, span_warning("Take the paddles out to continue."))
+		return FALSE
+	if(!ishuman(patient))
```

Comment:
this is fine, but just a note that when you're checking if something is the type that it is already typecaste as, you can just go istype(patient)

Reply:
personally prefer it this way since its explicit but w/e